src/plugins/data/public/autocomplete/providers/value_suggestion_provider.ts

```diff
@@ -39,7 +39,16 @@ function resolver(title: string, field: IFieldType, query: string, boolFilter: a
   return [ttl, query, title, field.name, JSON.stringify(boolFilter)].join('|');
 }
 
-export type ValueSuggestionsGetFn = (args: ValueSuggestionsGetFnArgs) => Promise<any[]>;
+export type ValueSuggestionsGetFn = (
+  args: ValueSuggestionsGetFnArgs | ValueSuggestionsSQLGetFnArgs
+) => Promise<any[]>;
```

Comment:
np. might be better to put args type and return type as generic, to allow extending and keeps autocomplete generic to what languages are used

Reply:
I'm not sure I fully understand what making the args or the return type generic would accomplish. Since this file is meant to include the implementation for functions that would do the value completion, they require specific inputs that may not be related to the output. 
To keep autocomplete generic to the languages used, should I just change the name of the function to move away from "SQL"?